src/docs/framework-integration/vue.md

```diff
@@ -8,50 +8,378 @@ contributors:
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./lib",
+    "lib": ["dom", "es2020"],
+    "module": "es2015",
+    "moduleResolution": "node",
+    "target": "es2017",
+    "skipLibCheck": true
+  },
+  "include": ["src"],
+  "exclude": ["node_modules"]
 }
 ```
 
-Vue provides several different ways to install and use the framework in an application. The above technique for integrating a Stencil custom element library has been tested on a Vue application that was created using the `vue-cli` with ES2015 and WebPack as primary options. A similar technique should work if the application was generated using other options.
+Update your `package.json`, adding the following options to the existing config:
+
+```json
+{
+  "types": "lib/index.d.ts",
+  "scripts": {
+    "test": "echo \"Error: run tests from root\" && exit 1",
+    "build": "npm run tsc",
+    "tsc": "tsc -p ."
+  },
+  "publishConfig": {
+    "access": "public"
+  }
+}
+```
+
+### Adding Vue Output Target
+
+Install the `@stencil/vue-output-target` dependency to your Stencil component library package.
+
+```bash
+# Install dependency (from packages/stencil-library)
+npm install @stencil/vue-output-target --save-dev
+# of if you are using yarn
+yarn add @stencil/vue-output-target --dev
+```
+
+In your project's `stencil.config.ts`, add the `vueOutputTarget` configuration to the `outputTargets` array:
+
+```ts
+import { vueOutputTarget } from '@stencil/vue-output-target';
+
+export const config: Config = {
+  namespace: 'stencil-library',
+  outputTargets: [
+    vueOutputTarget({
+      componentCorePackage: 'your-stencil-library-package-name', // i.e.: stencil-library
+      proxiesFile: '../vue-library/src/components.ts',
+    }),
+  ],
+};
+```
+
+> The `proxiesFile` is the relative path to the file that will be generated with all the Vue component wrappers. You will replace the file path to match your project's structure and respective names. You can generate any file name instead of `components.ts`.
+
+You can now build your Stencil component library to generate the component wrappers.
+
+```bash
+# Build the library and wrappers (from packages/stencil-library)
+npm run build
+# or if you are using yarn
+yarn run build
+```
+
+If the build is successful, you will not have contents in the file specifies in `proxiesFile`.
+
+#### Vue Plugin
+
+To register your web components for the lazy-loaded (hydrated) bundle, you will need to create a new file for the Vue plugin:
+
+```ts
+// packages/vue-library/src/plugin.ts
+
+import { Plugin } from 'vue';
+import { applyPolyfills, defineCustomElements } from 'stencil-library/loader';
+
+export const ComponentLibrary: Plugin = {
+  async install() {
+    applyPolyfills().then(() => {
+      defineCustomElements();
+    });
+  },
+};
+```
+
+This plugin will be used by Vue applications as follows:
+
+```ts
+import { ComponentLibrary } from 'vue-library';
+
+createApp(App).use(ComponentLibrary).mount('#app');
+```
```

Comment:
Does it make sense to have this snippet here? We repeat it not too much longer in the "Consumer Usage" section [here](https://github.com/ionic-team/stencil-site/pull/864/files#diff-af403489927c5c8dffa93fe5aa0478c28408f6b160f7e3e029475f676a005321R229-R236)